src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs

```diff
@@ -44,16 +46,21 @@ private static void OnCurrentPageViewModelChanged(DependencyObject d, Dependency
         if (d is SearchBar @this
             && e.NewValue is PageViewModel page)
         {
-            // TODO: In some cases we probably want commands to clear a filter somewhere in the process, so we need to figure out when that is.
+            // TODO: In some cases we probably want commands to clear a filter
+            // somewhere in the process, so we need to figure out when that is.
             @this.FilterBox.Text = page.Filter;
+            @this.FilterBox.Select(@this.FilterBox.Text.Length, 0);
```

Comment:
what is the @ decorator for?

Reply:
`this` is a keyword in C# normally, but this is a static method as it's the callback from the dependency property where we don't have the specific instance of our class, that's passed in as the first parameter, so we do a cast on line 46 and make our own reference to "this". But since it's a keyword, we use the `@` declarator to tell C# that we really want to use this as our variable name, basically it's a C# syntax feature to let you use keywords like `this`, `class`, etc... as variable names.

It's not that common, and probably not the best to use in many cases, but here is kind of a special case as we really are trying to refer to our own class here, we're just in a static method.

However, it's really just temporary, as when the DependencyProperty source generators come out of Toolkit Labs, then the DP declaration here can all be replaced with a nice source generator which will call a method for us on our class instance, so we'll already have our regular context and won't need `this` at all. 😊